### PR TITLE
Updated Readme

### PR DESCRIPTION
**Pull request by kapitoshka438**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,102 @@
...
+
+```
+cd docs-src
+../deps/bin/hugo server -D
+```
+
+### Tagging New Versions
+
+This repository has several components, and they need more than just a "vX.Y.Z"
+tag on the git repo.  In particular, this repository has two go modules, one of
```
Comment (mplooster) [inline README.md:55]: Is this referring to the examples go module and the gRPC go module? If so, we will need to remember to update this section when the examples are put in their own repository.

File: README.md [hunk 2]
```diff
@@ -1 +1,102 @@
...
+Step 3: Add version tags to the sources.
+
+```
+NEW_VERSION="1.0.1"
+
+git checkout master
+git checkout -b version-update-v$NEW_VERSION
+
+sed -i 's|DiathekeVersion = "[0-9.]*"|DiathekeVersion = "'$NEW_VERSION'"|g' grpc/swift-diatheke/Diatheke.swift
+sed -i 's|.upToNextMajor(from: "[0-9.]*")|.upToNextMajor(from: "'$NEW_VERSION'")|g' docs-src/content/using-diatheke-sdk/include.md
```
Comment (mplooster) [inline README.md:86]: This will not actually update the public-facing documentation unless hugo is also run.

File: README.md (hunk 1 above)
Comment (kapitoshka438) [reply README.md:55]: Honestly, I didn't look deeper on what is in other language modules. As I know, currently we don't have separate examples repos for Diatheke like for Cubic. Please, make a suggestion what else I should add to make version tagging work properly.

File: README.md (hunk 2 above)
Comment (kapitoshka438) [reply README.md:86]: After updating the docs-src should who or what should update the public documentation? It this responsibility of internal-tools?

File: README.md [hunk 3]
```diff
@@ -1 +1,102 @@
...
+The `grpc` folder contains auto-generated code in several languages.  In order
+to generate the code again, you should run `make`.  Generated code is checked
+in, and you must make sure it is up to date when you push commits to this
+repository.
+
+Code generation has the following dependencies:
+  - The protobuf compiler itself (protoc)
+  - The protobuf documentation generation plugin (protoc-gen-doc)
+  - The golang plugins (protoc-gen-go and protoc-gen-grpc-gateway)
+  - The python plugins (grpcio-tools and googleapis-common-protos)
```
[the remark this answers is not in the document]
Comment (kapitoshka438) [reply README.md:25]: You are right. I have added it.

File: README.md [hunk 4]
```diff
@@ -1 +1,102 @@
...
+  - The static website generator (hugo)
+
+A few system dependencies are required:
```
[the remark this answers is not in the document]
Comment (kapitoshka438) [reply README.md:29]: Added swift requirements

File: README.md (hunk 2 above)
Comment (mplooster) [reply README.md:88]: That is a great question. Currently, this repo doesn't have a Jenkins build set up  yet, so for now it is the responsibility of the person making changes to docs-src to run hugo and commit the generated docs.

File: README.md (hunk 1 above)
Comment (mplooster) [reply README.md:57]: My preference would be to move the example code and only have a single version on this repo. Perhaps @renaissanceGeek or @happyalu have opinions?

Comment (renaissanceGeek) [reply README.md:57]: I think that makes sense. It's better for the SDK to have a single version for all languages.  Ideally, when we tag a new SDK version, CI should compile the examples and flag breaking changes.

**Pull request merged**